### DISCUSSION TITLE
More efficient backtesting

### DISCUSSION THREAD

**Discussion by dszokolics** [alan-turing-institute/sktime#7214] [Ideas]
I'm running a backtest for a single time series with multiple years, with prediction horizon of 71-77. My main problem is that the runtime is extremely long. I have the following two considerations:
- I only have a single time series, so I'd like to check the model for as many periods as possible, which means that I predict for five years of data with one-week expanding windows.
- My prediction horizon is long, so recursive strategy is not working properly.

As I understand correctly, the backtest build one model for each single prediction. With 5 years of data it means 1.8k models built, which is fairly inefficient.

For models with this long prediction horizon, it doesn't really matter if we use lags from 71 or 77 periods earlier. Predicting the full horizon with a single model that has lags starting from 77 would have practically the same prediction performance, but the runtime would decrease sevenfold. Based on this, it would be nice to have a forecast strategy that builds a single model only with those lags that are available for all the horizons.

I'm quite new to sktime, but so far I haven't seen any alternatives how I can make my backtests run faster with the available tooling. I hope that somebody can suggest an already existing solution!

**Comment by fkiraly** [comment]
I think there are two possible answers here, and related issues:

1. one way to avoid multiple models to be constructed is to use `strategy="update"` and `strategy="no-update-params"`. Both strategies avoid creation of new instances, and `"no-update-params"` also avoids any update-fitting. Related: https://github.com/sktime/sktime/issues/7201 - Though refitting may happen at every `update`, if no efficient `update` is implemented.

2. Many models that could implement efficient update steps do not. This is an implementation task, an the reducer forecaster in particular could support more efficient updates that avoid refitting. It would be great if someone could help out implementing his for the reducers - see https://github.com/sktime/sktime/issues/7187. FYI @j-adamczyk who wanted to help with this as well.

3. There are a few online update wrappers which allow one to control how often an update is performed. The default is at every new batch, but using these one can set it to perform a model update only after a certain time: https://github.com/sktime/sktime/issues/7187 (`UpdateEvery` etc). However, this also hinges on `update` being implemented properly, for some forecasters.
